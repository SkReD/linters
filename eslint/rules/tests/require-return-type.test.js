const {CLIEngine} = require('eslint');
const path = require('path');

const cli = new CLIEngine({
    envs: ['browser', 'node'],
    useEslintrc: false,
    configFile: path.join(__dirname, '../../common/typescript/enviroment.js'),
    parserOptions: {
        createDefaultProgram: true,
    },
    rules: {
        '@tinkoff/require-return-type': 'error',
    },
    ignore: false,
});

const codeframe = cli.getFormatter('codeframe');

test('require-return-type snapshot', () => {
    const report = cli.executeOnFiles([
        path.join(__dirname, './require-return-type.test.js'),
        path.join(__dirname, './require-return-type.fixture.ts'),
    ]);

    expect(codeframe(report.results)).toMatchSnapshot();
});
