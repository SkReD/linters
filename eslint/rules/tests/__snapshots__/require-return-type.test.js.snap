// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`require-return-type snapshot 1`] = `
"[31merror[39m: [1mFunction with return keyword should specify return type[22m [2m(@tinkoff/require-return-type)[22m at [32meslint/rules/tests/require-return-type.fixture.ts:28:6[39m:
  26 |     }
  27 | 
> 28 |     c() {
     |      ^
  29 |         // Should fail
  30 |         return '123';
  31 |     }


[31merror[39m: [1mFunction with return keyword should specify return type[22m [2m(@tinkoff/require-return-type)[22m at [32meslint/rules/tests/require-return-type.fixture.ts:32:7[39m:
  30 |         return '123';
  31 |     }
> 32 |     c2(/* 123 */) {
     |       ^
  33 |         // Should fail
  34 |         return '123';
  35 |     }


[31merror[39m: [1mFunction with return keyword should specify return type[22m [2m(@tinkoff/require-return-type)[22m at [32meslint/rules/tests/require-return-type.fixture.ts:36:7[39m:
  34 |         return '123';
  35 |     }
> 36 |     c3() /* 123 */ {
     |       ^
  37 |         // Should fail
  38 |         return '123';
  39 |     }


[31merror[39m: [1mFunction with return keyword should specify return type[22m [2m(@tinkoff/require-return-type)[22m at [32meslint/rules/tests/require-return-type.fixture.ts:40:7[39m:
  38 |         return '123';
  39 |     }
> 40 |     c4() {
     |       ^
  41 |         // 321 // Should fail
  42 |         return '123';
  43 |     }


[31merror[39m: [1mFunction with return keyword should specify return type[22m [2m(@tinkoff/require-return-type)[22m at [32meslint/rules/tests/require-return-type.fixture.ts:87:1[39m:
  85 | }
  86 | 
> 87 | function ab3() {
     | ^
  88 |     // Should fail
  89 |     return '123';
  90 | }


[31m[1m5 errors found.[22m[39m"
`;
